test/helpers/position_assertions.cc

```diff
@@ -763,6 +769,39 @@ string TypeAssertion::toString() const {
     return fmt::format("type: {}", symbol);
 }
 
+FileDefAssertion::FileDefAssertion(std::string_view filename, std::unique_ptr<Range> &range, int assertionLine,
+                                   std::string targetFilename, int targetLine, int targetColumn)
+    : BaseDefAssertion(filename, range, assertionLine, ""sv), targetFilename(targetFilename), targetLine(targetLine),
+      targetColumn(targetColumn) {}
+
+std::shared_ptr<FileDefAssertion> FileDefAssertion::make(std::string_view filename, std::unique_ptr<Range> &range,
+                                                         int assertionLine, std::string_view assertionContents,
+                                                         std::string_view assertionType) {
+    vector<string_view> split = absl::StrSplit(assertionContents, ' ');
+    CHECK_EQ(split.size(), 3);
+    {
+        INFO(fmt::format("Invalid file-def assertion; multiple words found:\n{}\nFile definition assertions should be "
+                         "of the form:\n# [^*] file-def: file-path line column",
+                         assertionContents));
+    }
+    auto file_path = split[0];
+    auto line = stoi(string(split[1]));
+    auto col = stoi(string(split[2]));
+
+    return make_shared<FileDefAssertion>(filename, range, assertionLine, string(file_path), line, col);
+}
+
+std::unique_ptr<Location> FileDefAssertion::getDefinitionLocation(const LSPConfiguration &config) const {
+    auto targetPath = make_path(filename).replace_filename(targetFilename);
+    auto targetUri = filePathToUri(config, targetPath.string());
+    auto targetRange = RangeAssertion::makeRange(targetLine, targetColumn, targetColumn + 1);
+    return make_unique<Location>(targetUri, targetRange->copy());
+}
+
+string FileDefAssertion::toString() const {
+    return fmt::format("file: {}:{}+{}", targetFilename, targetLine, targetColumn);
```

Comment:
aren't they separated by spaces?

```suggestion
    return fmt::format("file-def: {} {} {}", targetFilename, targetLine, targetColumn);
```